### PR TITLE
[CTOR-38][Plugin] New pack Avigilon camera snmp

### PR DESCRIPTION
**Pull request by lucie-dubrunfaut**
## Description

New pack Avigilon-Camera-SNMP monitoring:
- uptime
- interfaces
- storage
- memory
- temperature

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

<h2> How this pull request can be tested ? </h2>

Using robot tests

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by kduret — DISMISSED** [review]

File: tests/robot/hardware/devices/camera/avigilon/snmp/storage.robot [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    ${output}    Strip String    ${output}
+    Should Be Equal As Strings
+    ...    ${output}
+    ...    ${expected_result}
+    ...    \nWrong output result for command:\n${command}\n\nExpected:\n${expected_result}\nCommand output:\n${output}\n\n
+
+    Examples:        tc    warning_status                               critical_status                             expected_result    --
+            ...      1     ${EMPTY}                                     ${EMPTY}                                    OK: state of the SD card: mediaNotPresent
+            ...      2     '\%{storage_state\} =~ /mediaNotPresent/'   ${EMPTY}                                    WARNING: state of the SD card: mediaNotPresent
+            ...      3     ${EMPTY}                                     '\%{storage_state\} =~ /mediaNotPresent/'  CRITICAL: state of the SD card: mediaNotPresent
```
Comment (omercier) [inline tests/robot/hardware/devices/camera/avigilon/snmp/storage.robot:37]:
```suggestion
            ...      2     '${PERCENT}\{storage_state\} =~ /mediaNotPresent/'   ${EMPTY}                                    WARNING: state of the SD card: mediaNotPresent
            ...      3     ${EMPTY}                                     '${PERCENT}\{storage_state\} =~ /mediaNotPresent/'  CRITICAL: state of the SD card: mediaNotPresent
```

Comment (omercier) [reply tests/robot/hardware/devices/camera/avigilon/snmp/storage.robot:37]: Try with the ${PERCENT} builtin variable (not even sure the brackets need to be escaped)

**Review by omercier — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Review by Evan-Adam — APPROVED** [review]

File: src/hardware/devices/camera/avigilon/snmp/mode/storage.pm [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        13 => 'error'
+    };
+
+    my $oid_storage_state = '.1.3.6.1.4.1.46202.1.1.1.5.0'; #storageState
+    my $snmp_result       = $options{snmp}->get_leef(
+        oids         => [$oid_storage_state],
+        nothing_quit => 1
+    );
+
+    if (defined($storage_state_mapping->{$snmp_result->{$oid_storage_state}}) && $storage_state_mapping->{$snmp_result->{$oid_storage_state}} ne ''){
```
Comment (Evan-Adam) [inline src/hardware/devices/camera/avigilon/snmp/mode/storage.pm:88]:
```suggestion
    if (is_empty($storage_state_mapping->{$snmp_result->{$oid_storage_state}}) ){
```

File: src/hardware/devices/camera/avigilon/snmp/mode/storage.pm [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+package hardware::devices::camera::avigilon::snmp::mode::storage;
+
+use base qw(centreon::plugins::templates::counter);
+use centreon::plugins::templates::catalog_functions qw(catalog_status_threshold_ng);
+
```
Comment (Evan-Adam) [inline src/hardware/devices/camera/avigilon/snmp/mode/storage.pm:25]:
```suggestion
use centreon::plugins::misc qw(is_empty);

```

**Pull request merged**